### PR TITLE
add consul-on-ubuntu template

### PR DESCRIPTION
**Pull request by esell**
This PR is to add a new template that spins up a 3 node [Consul](https://www.consul.io) cluster and puts it behind an internal LB. The pre-reqs (hence the .ci_skip) are an account with Hashicorpo's [Atlas](https://atlas.hashicorp.com/) otherwise auto-creation of the cluster is a bit messy.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@esell__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (euseller). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: consul-on-ubuntu/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+                "description": "Admin username for VM"
+            }
+        },
+        "adminPassword": {
+            "type": "securestring",
+            "metadata": {
+                "description": "Admin password for VM"
+            }
+        },
+        "newStorageAccountName": {
```
Comment (singhkay) [inline consul-on-ubuntu/azuredeploy.json]: See #944 about doing storage accounts in a non error-prone way

File: consul-on-ubuntu/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+            }
+        },
+        "vmSize": {
+            "type": "string",
+            "defaultValue": "Standard_A1",
+            "allowedValues": [
+                "Standard_A1",
+                "Standard_A2",
+                "Standard_A3",
+                "Standard_D1",
```
Comment (singhkay) [inline consul-on-ubuntu/azuredeploy.json]: How about D*_v2 sizes instead?

File: consul-on-ubuntu/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+                "Standard_D2",
+                "Standard_D3"
+            ],
+            "metadata": {
+                "description": "Size of the Virtual Machine."
+            }
+        },
+        "atlasInfra": {
+            "type": "string",
```
Comment (singhkay) [inline consul-on-ubuntu/azuredeploy.json]: Let's prefix these atlas infra parameters with 'existing' to set the right expectations

File: consul-on-ubuntu/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+        "subnet1Prefix": "10.0.0.0/24",
+        "subnet1Ref": "[concat(variables('vnetID'),'/subnets/', variables('subnet1Name'))]",
+        "publicIPAddressType": "Dynamic",
+        "vmStorageAccountContainerName": "vhds"
+    },
+    "resources": [
+        {
+            "type": "Microsoft.Storage/storageAccounts",
+            "name": "[parameters('newStorageAccountName')]",
+            "apiVersion": "2015-05-01-preview",
```
Comment (singhkay) [inline consul-on-ubuntu/azuredeploy.json]: Use the GA API versions 2015-06-15 and also make apiVersion a variable and reuse across resources below

File: consul-on-ubuntu/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,327 @@
...
+                "backendAddressPools": [
+                    {
+                        "name": "loadBalancerBackEnd"
+                    }
+                ],
+                "loadBalancingRules": [
+                    {
+                        "properties": {
+                            "frontendIPConfiguration": {
+                                "id": "[concat(resourceId('Microsoft.Network/loadBalancers', concat(parameters('appPrefix'),'_LB')), '/frontendIpConfigurations/loadBalancerFrontEnd')]"
```
Comment (singhkay) [inline consul-on-ubuntu/azuredeploy.json]: Can you format this JSON with 2-space scheme? Reading these deep nested objects with 4 space is wreaking havoc with the Github default width

File: consul-on-ubuntu/metadata.json [hunk 6]
```diff
@@ -0,0 +1,7 @@
+{
+  "itemDisplayName": "Deploys a 3 node Consul Cluster",
+  "description": "This template deploys a 3 node Consul cluster and auto-joins the nodes via Atlas",
```
Comment (singhkay) [inline consul-on-ubuntu/metadata.json]: Can you add a bit more to the description to describe what Consul and Atlas are?

**Comment by esell** [comment]
Ok, requested changes have been made :tada:

**Comment by esell** [comment]
shoot, sorry about that. my formatting fixed caused chaos over the entire file. let me re-work that and re-submit.

**Comment by esell** [comment]
ok, i _think_ it should all be cleaned up now.

File: consul-on-ubuntu/README.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+<a href="http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2Fconsul-on-ubuntu%2Fazuredeploy.json" target="_blank">
+  <img src="http://armviz.io/visualizebutton.png"/>
+</a>
+
+
+# Consul on Ubuntu
+
+This will create a 3 node [Consul](https://www.consul.io/) cluster in Azure and put them into an internal load balancer.
+
+**NOTE** you will need to have an [Atlas](https://atlas.hashicorp.com/) account and token in order for auto-join to work.
```
Comment (singhkay) [inline consul-on-ubuntu/README.md]: Is there a guide on how to setup Atlas infra and get the token?

**Comment by singhkay** [comment]
@esell Looks good mostly, just a minor nitpick :)

**Comment by esell** [comment]
I added the only documentation I could find but Atlas is a pretty basic app (from the UI perspective) so hopefully others won't run into any issues setting things up. I've honestly never used it (Atlas) until I built this template and it was pretty painless to setup. Up until now I had always scripted the consul join portion via other methods (salt, ansible, etc) but the goal for this template was to get something up and running with as little additional tooling as possible.

**Comment by singhkay** [comment]
lgtm, merged

**Pull request merged**